core/jvm/src/main/scala/bobcats/Hash1Platform.scala

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2021 Typelevel
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package bobcats
+
+import java.security.MessageDigest
+import scodec.bits.ByteVector
+import cats.Applicative
+import cats.effect.Sync
+import fs2.{Chunk, Stream}
+
+private final class JavaSecurityDigest[F[_]](val hash: MessageDigest)(
+    implicit F: Applicative[F])
+    extends UnsealedHash1[F] {
+
+  override def digest(data: ByteVector): F[ByteVector] = F.pure {
+    val h = hash.clone().asInstanceOf[MessageDigest]
```

Comment:
Doesn't look like cloning is always supported.

> Implementations are free to implement the Cloneable interface. Client applications can test cloneability by attempting cloning and catching the CloneNotSupportedException: 

https://docs.oracle.com/javase/8/docs/api/java/security/MessageDigest.html